core/src/debug/frameInfo.cpp

```diff
@@ -87,25 +105,48 @@ void FrameInfo::draw(const View& _view, TileManager& _tileManager) {
-        debuginfos.push_back("tile cache size:"
-                + std::to_string(_tileManager.getTileCache()->getMemoryUsage() / 1024) + "kb");
-        debuginfos.push_back("tile size:" + std::to_string(memused / 1024) + "kb");
-        debuginfos.push_back("avg frame cpu time:" + to_string_with_precision(avgTimeCpu, 2) + "ms");
-        debuginfos.push_back("avg frame render time:" + to_string_with_precision(avgTimeRender, 2) + "ms");
-        debuginfos.push_back("avg frame update time:" + to_string_with_precision(avgTimeUpdate, 2) + "ms");
-        debuginfos.push_back("zoom:" + std::to_string(_view.getZoom()));
-        debuginfos.push_back("pos:" + std::to_string(_view.getPosition().x) + "/"
-                + std::to_string(_view.getPosition().y));
-        debuginfos.push_back("tilt:" + std::to_string(_view.getPitch() * 57.3) + "deg");
-        debuginfos.push_back("pixel scale:" + std::to_string(_view.pixelScale()));
-
-        TextDisplay::Instance().draw(debuginfos);
-    }
+        if (getDebugFlag(DebugFlags::tangram_infos)) {
+            std::vector<std::string> debuginfos;
+
+            debuginfos.push_back("visible tiles:"
+                                 + std::to_string(_tileManager.getVisibleTiles().size()));
+            debuginfos.push_back("tile cache size:"
+                                 + std::to_string(_tileManager.getTileCache()->getMemoryUsage() / 1024) + "kb");
+            debuginfos.push_back("tile size:" + std::to_string(memused / 1024) + "kb");
+            debuginfos.push_back("avg frame cpu time:" + to_string_with_precision(avgTimeCpu, 2) + "ms");
+            debuginfos.push_back("avg frame render time:" + to_string_with_precision(avgTimeRender, 2) + "ms");
+            debuginfos.push_back("avg frame update time:" + to_string_with_precision(avgTimeUpdate, 2) + "ms");
+            debuginfos.push_back("zoom:" + std::to_string(_view.getZoom()));
+            debuginfos.push_back("pos:" + std::to_string(_view.getPosition().x) + "/"
+                                 + std::to_string(_view.getPosition().y));
+            debuginfos.push_back("tilt:" + std::to_string(_view.getPitch() * 57.3) + "deg");
+            debuginfos.push_back("pixel scale:" + std::to_string(_view.pixelScale()));
+
+            TextDisplay::Instance().draw(debuginfos);
+        }
 
+        if (getDebugFlag(DebugFlags::stats)) {
+            const int scale = 5 * _pixelsPerPoint;
+            int i = 0;
+
+            Primitives::setColor(0xfff000);
+            for (float t : cputime) {
+                i += 4 * _pixelsPerPoint;
+                Primitives::drawLine(glm::vec2(i, 0), glm::vec2(i, t * scale));
+            }
+
+            i = 2 * _pixelsPerPoint;
+            Primitives::setColor(0x0000ff);
+            for (float t : rendertime) {
+                i += 4 * _pixelsPerPoint;
+                Primitives::drawLine(glm::vec2(i, 0), glm::vec2(i, t * scale));
```

Comment:
Actually what we show currently for a frame is not really GPU time because:
- cputime: time taken in `tangram.cpp:render()`
- rendertime: time taken in `tangram.cpp:render()` + `glFinish()` so it encompasses all gl commands dispatched from from current frame.

We should stack them like proposed, but also have in `tangram.cpp:update()` in it so it's the whole tangram frame and the 16.6ms would make sense as a notice of 60fps.

I will update for this.